src/business_logic/transaction/objects/v2/declare_v2.rs

```diff
@@ -0,0 +1,330 @@
+use std::collections::HashMap;
+
+use crate::{
+    business_logic::{
+        execution::{
+            execution_entry_point::ExecutionEntryPoint,
+            objects::{CallInfo, TransactionExecutionContext, TransactionExecutionInfo},
+        },
+        fact_state::state::ExecutionResourcesManager,
+        state::state_api::{State, StateReader},
+        transaction::{
+            error::TransactionError,
+            fee::{calculate_tx_fee, execute_fee_transfer, FeeInfo},
+            objects::internal_invoke_function::verify_no_calls_to_other_contracts,
+        },
+    },
+    core::{
+        contract_address::v2::starknet_sierra_contract_address::compute_sierra_class_hash,
+        transaction_hash::starknet_transaction_hash::calculate_declare_v2_transaction_hash,
+    },
+    definitions::{
+        constants::VALIDATE_DECLARE_ENTRY_POINT_SELECTOR, general_config::StarknetGeneralConfig,
+        transaction_type::TransactionType,
+    },
+    services::api::contract_class::EntryPointType,
+    utils::{calculate_tx_resources, Address},
+};
+use cairo_lang_starknet::contract_class::ContractClass as SierraContractClass;
+use felt::Felt252;
+use num_traits::Zero;
+#[derive(Debug)]
+pub struct InternalDeclareV2 {
+    pub sender_address: Address,
+    pub tx_type: TransactionType,
+    pub validate_entry_point_selector: Felt252,
+    pub version: u64,
+    pub max_fee: u64,
+    pub signature: Vec<Felt252>,
```

Comment:
Is the signature variable in size? Otherwise, could this be an array?

Reply:
is not clear if is variable in size, i should check